modules/Emails/include/ListView/ListViewDataEmailsSearchOnCrm.php

```diff
@@ -0,0 +1,316 @@
+                $secondary_query = $ret_array['secondary_select'] . $ret_array['secondary_from'] . ' WHERE '.$this->lvde->seed->table_name.'.id IN ' .$id_list;
+                if(isset($ret_array['order_by']))
+                {
+                    $secondary_query .= ' ' . $ret_array['order_by'];
+                }
+
+                $secondary_result = $this->lvde->db->query($secondary_query);
+
+                $ref_id_count = array();
+                while($row = $this->lvde->db->fetchByAssoc($secondary_result)) {
+
+                    $ref_id_count[$row['ref_id']][] = true;
+                    foreach($row as $name=>$value) {
+                        //add it to every row with the given id
+                        foreach($idIndex[$row['ref_id']] as $index){
+                            $rows[$index][$name]=$value;
+                        }
+                    }
+                }
+
+                $rows_keys = array_keys($rows);
+                foreach($rows_keys as $key)
+                {
+                    $rows[$key]['secondary_select_count'] = count($ref_id_count[$rows[$key]['ref_id']]);
+                }
+            }
+
+            // retrieve parent names
+            if(!empty($filter_fields['parent_name']) && !empty($filter_fields['parent_id']) && !empty($filter_fields['parent_type'])) {
+                foreach($idIndex as $id => $rowIndex) {
+                    if(!isset($post_retrieve[$rows[$rowIndex[0]]['parent_type']])) {
+                        $post_retrieve[$rows[$rowIndex[0]]['parent_type']] = array();
+                    }
+                    if(!empty($rows[$rowIndex[0]]['parent_id'])) $post_retrieve[$rows[$rowIndex[0]]['parent_type']][] = array('child_id' => $id , 'parent_id'=> $rows[$rowIndex[0]]['parent_id'], 'parent_type' => $rows[$rowIndex[0]]['parent_type'], 'type' => 'parent');
+                }
+                if(isset($post_retrieve)) {
+                    $parent_fields = $seed->retrieve_parent_fields($post_retrieve);
+                    foreach($parent_fields as $child_id => $parent_data) {
+                        //add it to every row with the given id
+                        foreach($idIndex[$child_id] as $index){
+                            $rows[$index]['parent_name']= $parent_data['parent_name'];
+                        }
+                    }
+                }
+            }
+
+            $pageData = array();
+
+            reset($rows);
+            while($row = current($rows)){
+
+                $temp = clone $seed;
+                $dataIndex = count($data);
+
+                $temp->setupCustomFields($temp->module_dir);
+                $temp->loadFromRow($row);
+                if (empty($this->lvde->seed->assigned_user_id) && !empty($temp->assigned_user_id)) {
+                    $this->lvde->seed->assigned_user_id = $temp->assigned_user_id;
+                }
+                if($idIndex[$row[$idField]][0] == $dataIndex){
```

Comment:
(class ListViewDataEmailsSearchOnIMap and class ListViewDataEmailsSearchOnCrm not completely refactored yet)